spikeinterface/core/base.py

```diff
@@ -256,6 +256,10 @@ def copy_metadata(self, other, only_main=False, ids=None):
         # TODO: copy features also
 
         other.extra_requirements.extend(self.extra_requirements)
+        
+        if hasattr(self, "preferred_mp_context"):
+            if self.preferred_mp_context is not None:
+                other.preferred_mp_context = self.preferred_mp_context    
```

Comment:
I think this kind of attribute transmited here is a bit dangerous because it is not dumpable.
Are we sure that on load back we keep this flag ? Maybe we could put this in annotation with`__` naming.
Also I would vote for having this attribute = None always to avoid this `hasattr` no ?